docs/iris/src/further_topics/lenient_metadata.rst

```diff
@@ -0,0 +1,477 @@
+.. table:: - :ref:`Strict equality <metadata equality>`
+   :widths: auto
+   :align: center
+
+   ======== ======== =========
+   left     right    **equal**
+   ======== ======== =========
+   ``X``    ``Y``    ``False``
+   ``Y``    ``X``    ``False``
+   ``X``    ``X``    ``True``
+   ``X``    ``None`` ``False``
+   ``None`` ``X``    ``False``
+   ======== ======== =========
+
+.. _strict difference table:
+.. table:: - :ref:`Strict difference <metadata difference>`
+   :widths: auto
+   :align: center
+
+   ======== ======== =================
+   left     right    **difference**
+   ======== ======== =================
+   ``X``    ``Y``    (``X``, ``Y``)
+   ``Y``    ``X``    (``Y``, ``X``)
+   ``X``    ``X``    ``None``
+   ``X``    ``None`` (``X``, ``None``)
+   ``None`` ``X``    (``None``, ``X``)
+   ======== ======== =================
+
+.. _strict combine table:
+.. table:: - :ref:`Strict combination <metadata combine>`
+   :widths: auto
+   :align: center
+
+   ======== ======== ===========
+   left     right    **combine**
+   ======== ======== ===========
+   ``X``    ``Y``    ``None``
+   ``Y``    ``X``    ``None``
+   ``X``    ``X``    ``X``
+   ``X``    ``None`` ``None``
+   ``None`` ``X``    ``None``
+   ======== ======== ===========
+
+.. _strict example:
+
+This type of **strict** behaviour does offer obvious benefit and value. However,
+it can be unnecessarily restrictive. For example, consider the metadata of the
+following ``latitude`` coordinate,
+
+.. doctest:: strict-behaviour
+
+    >>> latitude.metadata
+    DimCoordMetadata(standard_name='latitude', long_name=None, var_name='latitude', units=Unit('degrees'), attributes={}, coord_system=GeogCS(6371229.0), climatological=False, circular=False)
+
+Now, let's create a doctored version of this metadata with a different ``var_name``,
+
+.. doctest:: strict-behaviour
+
+    >>> metadata = latitude.metadata._replace(var_name=None)
+    >>> metadata
+    DimCoordMetadata(standard_name='latitude', long_name=None, var_name=None, units=Unit('degrees'), attributes={}, coord_system=GeogCS(6371229.0), climatological=False, circular=False)
+
+Clearly, these metadata are different,
+
+.. doctest:: strict-behaviour
+
+    >>> metadata != latitude.metadata
+    True
+    >>> metadata.difference(latitude.metadata)
+    DimCoordMetadata(standard_name=None, long_name=None, var_name=(None, 'latitude'), units=None, attributes=None, coord_system=None, climatological=None, circular=None)
+
+And yet, they both have the same ``name``, which some may find slightly confusing
+(see :meth:`~iris.common.metadata.BaseMetadata.name` for clarification)
+
+.. doctest:: strict-behaviour
+
+    >>> metadata.name()
+    'latitude'
+    >>> latitude.name()
+    'latitude'
+
+Regardlessly, at the end of the day we may not actually care that the
+``var_name`` is different. However, Iris relentlessly forces us to deal
+with such a difference; sometimes this can be challenging to overcome.
```

Comment:
I think you might mean "Regardless"? `Regardlessly` is an adverb so it doesn't make sense to be used in this way. Also, `At end of the day` is very informal language. I suppose it is in keeping with the tone of the rest of the metadata documentation, but I personally find that phrase, in particular, is a bit too informal. It may be fine in a email or in conversation, but not in official documentation. 

`Iris relentlessly forces us to deal with such a difference; sometimes this can be challenging to overcome.`
I find this way too critical of Iris, not too mention unfair. If I was a user and I read this, I would think "Why should I use a package that even the developer's are happily criticising in their own documentation!"
I do get the point you're trying to make, but it makes Iris sounds like too much of a bother to use, so I think it needs to be rephrased in a different way.

Maybe something like:
Regardless of whether the difference in var_name matters, Iris will treat the metadata as different. Thus the user is required to manually modify the metadata to avoid this.